watertap/tools/parallel/parallel_manager.py

```diff
@@ -0,0 +1,148 @@
+from abc import abstractmethod, ABC
+import os
+
+import numpy
+
+from pyomo.common.dependencies import attempt_import
+
+mpi4py, mpi4py_available = attempt_import("mpi4py")
+
+
+class ParallelManager(ABC):
+    @classmethod
+    def create_parallel_manager(cls, parallel_manager_class=None):
+        """
+        Create and return an instance of a ParallelManager, based on the libraries available in the
+        runtime environment.
+
+        Allows an optional python class to be passed in as parallel_manager_class. If so, this class
+        is instantiated and returned rather than checking the local environment.
+        """
+        if parallel_manager_class is not None:
+            return parallel_manager_class()
+
+        if mpi4py_available and ParallelManager.has_mpi_peer_processes():
```

Comment:
```suggestion
        if ParallelManager.has_mpi_peer_processes():
```
I guess we needn't check twice.